sql/atxs/atxs.go

```diff
@@ -891,6 +891,38 @@ func Units(db sql.Executor, atxID types.ATXID, nodeID types.NodeID) (uint32, err
 	return units, err
 }
 
+// ContributedToAtxInEpoch returns the ID of the ATX that the given identity contributed PoST to in the given epoch.
+//
+// It works by finding an ATX in the given epoch that has a PoST contribution from the given identity.
+// - `epoch` is looked up in the `atxs` table by matching atxid.
+func ContributedToAtxInEpoch(db sql.Executor, nodeID types.NodeID, epoch types.EpochID) (types.ATXID, error) {
+	var id types.ATXID
+	rows, err := db.Exec(`
+		SELECT atxid
+		FROM posts
+		WHERE pubkey = ?1 AND atxid IN (
+			SELECT id
+			FROM atxs
+			WHERE epoch = ?2
+		);`,
```

Comment:
This query appears correct, but that might be a bit better perhaps?
```suggestion
		SELECT p.atxid
		FROM posts p
		INNER JOIN atxs a ON p.atxid = a.id
		WHERE p.pubkey = ?1 AND a.epoch = ?2`),
```
Efficiency-wise I think it'll be roughly same

Reply:
Thanks!